flexmeasures/ui/templates/crud/asset.html

```diff
@@ -347,6 +458,437 @@ <h3>All child assets for {{ asset.name }}</h3>
+        const response = await fetch(apiUrl);
+        const asset = await response.json();
+    
+        localStorage.setItem(cacheKey, JSON.stringify(asset));
+    
+        return asset;
+    }
+    
+    // Fetch Sensor Details
+    async function getSensor(id) {
+        const cacheKey = `sensor_${id}`;
+        const cachedData = localStorage.getItem(cacheKey);
+    
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+    
+        const apiUrl = apiBasePath + "/api/v3_0/sensors/" + id;
+        const response = await fetch(apiUrl);
+        const sensor = await response.json();
+    
+        localStorage.setItem(cacheKey, JSON.stringify(sensor));
+    
+        return sensor;
+    }
+    
+    // highlight selected graph 
+    async function selectGraph(graphIndex) {
+        if (graphIndex !== undefined) {
+            savedGraphIndex = graphIndex;
+            selectedGraphTitle = sensorsToShow[graphIndex]?.title;
+            // check if graphIndex still exists(This is because this function is called when the removed button is clicked as well)
+            if (sensorsToShow[graphIndex]) {
+                renderApiSensors(cachedFilteredSensors, graphIndex);
+            } else {
+                renderApiSensors(cachedFilteredSensors);
+            }
+        } else {
+            savedGraphIndex = undefined;
+            selectedGraphTitle = undefined;
+            renderGraphCards();
+        }
+    }
+    
+    // Function to render the graph cards
+    async function renderGraphCards() {
+        const graphList = document.getElementById("graphList");
+        graphList.innerHTML = "";
+        const newSensorsToShow = [];
+    
+        if (sensorsToShow.length === 0) {
+            return;
+        }
+    
+        for (const [index, item] of sensorsToShow.entries()) {
+            const col = document.createElement("div");
+            col.classList.add("col-12", "mb-1");
+    
+            const sensorsUnits = [];
+            // the initializing of the newItem is to handle the case where the item is an array of ID's instead of an object
+            const newItem = {
+                title: item.title ?? "No Title",
+                sensors: item.sensors ?? (Array.isArray(item) ? item : [item]),
+            }
+            newSensorsToShow.push(newItem);
+    
+            const sensorsContent = await Promise.all(
+                newItem.sensors.map(async (sensor, sensorIndex) => {
+                    const sensorData = await getSensor(sensor);
+                    const Asset = await getAsset(sensorData.generic_asset_id);
+                    const Account = await getAccount(Asset.account_id);
+                    sensorsUnits.push(sensorData.unit);
+    
+                return `
+                    <div class="p-1 mb-3 border-bottom border-secondary">
+                        <div class="d-flex justify-content-between align-items-center">
```

Comment:
If we don't use `align-items-center` then the "X" will be more top right than just top centered, which I find nicer. Do you see a problem with that?

Reply:
No, that's ok. Ill modify it.